crates/controller/src/hud/details.rs

```diff
@@ -1,37 +1,130 @@
 use bevy::prelude::*;
+use de_core::baseset::GameSet;
 use de_core::{cleanup::DespawnOnGameExit, gamestate::GameState};
+use de_energy::Battery;
+use de_gui::{BodyTextCommands, BodyTextOps, GuiCommands, OuterStyle};
 
 use super::{interaction::InteractionBlocker, HUD_COLOR};
+use crate::selection::Selected;
+
+const PREFIXES: [&str; 5] = ["T", "G", "M", "k", ""];
 
 pub(crate) struct DetailsPlugin;
 
 impl Plugin for DetailsPlugin {
     fn build(&self, app: &mut App) {
-        app.add_system(spawn_details.in_schedule(OnEnter(GameState::Playing)));
+        app.add_system(setup.in_schedule(OnEnter(GameState::Playing)))
```

Comment:
Please add a cleanup system which removes `DetailsText` after the game is finished.